lib/puppet/pops/loader/loader_paths.rb

```diff
@@ -221,6 +226,20 @@ def relative_path
       TASKS_PATH
     end
 
+    def typed_name(type, name_authority, relative_path, module_name)
+      n = ''
+      n << module_name unless module_name.nil?
+
+      # Remove extension regardless of what it is. A task name cannot contain dots
+      relative_path = relative_path.sub(/\.[^\/]*\z/, '')
+
+      relative_path.split('/').each do |segment|
```

Comment:
This isn't safe on Windows. I'd recommending using `Pathname.ascend` instead.

Reply:
@joshcooper, please note though, that all relative paths in this context stem from use of Pathname. The origin of the path is the method `FileBased#relative_paths`.